File: spring-cloud-square-demo/consumer-retrofit-webflux/src/main/java/com/livk/consumer/RemoteService.java
package com.livk.consumer;

import org.springframework.cloud.square.retrofit.core.RetrofitClient;
import reactor.core.publisher.Mono;
import retrofit2.http.GET;

/**
 * <p>
 * RemoteService
 * </p>
 *
 * @author livk
 * @date 2022/4/13
 */
@RetrofitClient("square-provider")
public interface RemoteService {

    @GET("instance")
    Mono<String> instance();

}

File: spring-cloud-square/consumer-retrofit-webflux/src/main/kotlin/com/livk/consumer/RemoteService.kt
package com.livk.consumer

import org.springframework.cloud.square.retrofit.core.RetrofitClient
import reactor.core.publisher.Mono
import retrofit2.http.GET

/**
 * @author livk
 */
@RetrofitClient("square-provider")
interface RemoteService {
    @GET("instance")
    fun instance(): Mono<String>
}
